webapp-django/crashstats/api/views.py

```diff
@@ -328,12 +277,8 @@ def model_wrapper(request, model_name):
     return result, headers
 
 
-@pass_default_context
-def documentation(request, default_context=None):
-    context = default_context or {}
-
-    endpoints = []
-
+def api_models_and_names():
```

Comment:
``api_models_and_names`` extracts the code that generates the list of API models, so that they can be tested directly outside of a request. The list is before any user permissions are checked, so that the code can test all API models, rather than have bugs that only appear in production for privileged users.

Reply:
+1